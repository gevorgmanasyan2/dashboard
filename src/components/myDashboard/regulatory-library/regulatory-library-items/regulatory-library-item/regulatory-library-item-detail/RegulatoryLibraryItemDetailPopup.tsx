import React, { FC } from 'react'
import Popup from '../../../../../Popup/Popup'
import WaiverItem from './waiver-item/waiverItem'

interface RegDetailProps {
  waiverLibraryItems?: object[]
  closeDetailModal: () => void
  visible: boolean
}

const RegulatoryLibraryItemDetailPopup: FC<RegDetailProps> = ({
  waiverLibraryItems,
  closeDetailModal,
  visible,
}) => {

  return (
    <Popup
      className="waiver__item__container__modal"
      visible={visible}
      closeModal={closeDetailModal}
      modalTitle="Resource Library Details"
      footer={null}
    >
      {waiverLibraryItems &&
        waiverLibraryItems.map((item: any) => (
          <WaiverItem
            key={item.id}
            name={item.name}
            description={item.description}
            imageSource={item.image_source}
            actionLink={item.action_link}
            fileName={item.download_file}
            externalLink={item.external_link}
          />  
        ))}
    </Popup>
  )
}

export default RegulatoryLibraryItemDetailPopup
