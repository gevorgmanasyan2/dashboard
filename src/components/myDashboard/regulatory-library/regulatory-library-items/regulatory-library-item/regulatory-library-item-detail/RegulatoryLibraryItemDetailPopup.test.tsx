/* eslint-disable @typescript-eslint/no-empty-function */
/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import RegulatoryLibraryItemDetailPopup from './RegulatoryLibraryItemDetailPopup'

it('should render RegulatoryLibraryItemDetailPopup component', () => {
  const mockItem = {
    closeDetailModal: ()=>{},
    visible: true,
    waiverLibraryItems: [{ id: 1, name: 'test', waiverLibraryItems: [] }],
  }
  expect(shallow(<RegulatoryLibraryItemDetailPopup {...mockItem} />)).toMatchSnapshot()
})