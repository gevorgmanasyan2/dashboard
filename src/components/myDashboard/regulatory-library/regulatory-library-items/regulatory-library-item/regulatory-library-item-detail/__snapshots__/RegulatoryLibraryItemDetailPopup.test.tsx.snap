// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RegulatoryLibraryItemDetailPopup component 1`] = `
<Popup
  className="waiver__item__container__modal"
  closeModal={[Function]}
  footer={null}
  modalTitle="Resource Library Details"
  visible={true}
>
  <WaiverItem
    key="1"
    name="test"
  />
</Popup>
`;
